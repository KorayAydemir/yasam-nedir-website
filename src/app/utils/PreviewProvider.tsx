import LiveQueryProvider from "@sanity/preview-kit";
import { clientForPreview } from "../../sanity/clients";

const PreviewProvider = ({ children }: React.PropsWithChildren) => {
    const previewMode = process.env.SANITY_PREVIEW_ENABLED === "true";

    return !previewMode ? (
        children
    ) : (
        <LiveQueryProvider
            cache={{ includeTypes: ["settings"] }}
            logger={console}
            client={clientForPreview}
            token={process.env.SANITY_TOKEN_LIVE_PREVIEW}
            refreshInterval={1000 * 60}>
            {children}
        </LiveQueryProvider>
    );
};

export default PreviewProvider;
