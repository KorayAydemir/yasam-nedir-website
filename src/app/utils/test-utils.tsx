import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { setupStore } from "../../store";
import "@testing-library/jest-dom";

export const AllProviders = ({
    children,
}: {
    children: React.ReactElement;
}) => {
    const store = setupStore({});
    return (
        <Provider store={store}>
            <BrowserRouter>{children}</BrowserRouter>
        </Provider>
    );
};
