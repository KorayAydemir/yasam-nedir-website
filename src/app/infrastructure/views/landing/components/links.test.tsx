import renderer from "react-test-renderer";
import Links from "./links";
import { AllProviders } from "../../../../utils/test-utils";

test("snapshot test", () => {
    expect.assertions(1);

    const comp = renderer.create(
        <AllProviders>
            <Links />
        </AllProviders>
    );

    expect(comp).toMatchSnapshot();
});
