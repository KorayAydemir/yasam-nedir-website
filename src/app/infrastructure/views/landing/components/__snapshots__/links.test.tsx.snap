// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
[
  <a
    href="/yasambilim"
    onClick={[Function]}
  >
    Yaşam Bilim
  </a>,
  <a
    href="/evrim-kurami"
    onClick={[Function]}
  >
    Evrim Kuramı
  </a>,
  <a
    href="/denemeler"
    onClick={[Function]}
  >
    Denemeler
  </a>,
]
`;
